subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultMavenArtifactRepository.java

```diff
@@ -118,6 +118,9 @@ public void useGradleMetadata() {
     }
 
     protected boolean isPreferGradleMetadata() {
+        if (System.getProperty("org.gradle.internal.preferGradleMetadata") != null) {
```

Comment:
Why not just use the flag defined on the Maven repository?

Reply:
I wanted an easy way to execute the same build with and without Gradle metadata being used. I guess I could use an init script to modify all maven repositories, or modify the build script. But this seemed a simple solution to enable testing.

Reply:
:x: I think we should get rid of this property. We already have on magical thing with `useGradleMetadata()`. I think we should add that to the build script in the test setup. If that does not work in a generic way for all tests, it would not hurt to do it individually in a few tests.

The `setup()` is here in #3399: https://github.com/gradle/gradle/pull/3399/files#diff-daa95c5f363f3782808a33de727a571eR29

Can't we change that to something like:
```
executer.beforeExecute {
  if (publishModuleMetadata) {
    buildFile << """
      repositories.withType(MavenArtifactRepository) { it.useGradleMetadata() }
    """
  }
}
```